### PR TITLE
[IOTDB-5]Enable deletion in master

### PR DESCRIPTION
**Pull request by jt2594838**

### CODE REVIEW

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java [hunk 1]
```diff
@@ -467,9 +471,9 @@ public void update(String deviceId, String measurementId, long startTime, long e
       try {
         if (IoTDBDescriptor.getInstance().getConfig().enableWal) {
           overflowProcessor.getLogNode()
-                  .write(
-                          new UpdatePlan(startTime, finalEndTime, v, new Path(deviceId
-                                  + "." + measurementId)));
+              .write(
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: The line feed is a little weird, considering following style:
```
overflowProcessor.getLogNode().write(
     new UpdatePlan(startTime,  ...
```

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java [hunk 2]
```diff
@@ -641,11 +680,11 @@ public void endQuery(String deviceId, int token) throws FileNodeManagerException
...
    *
    * @param fileNodeName the seriesPath of storage group
-   * @param appendFile   the appended tsfile information
+   * @param appendFile the appended tsfile information
    */
   public boolean appendFileToFileNode(String fileNodeName, IntervalFileNode appendFile,
-                                      String appendFilePath)
-          throws FileNodeManagerException {
+      String appendFilePath)
+      throws FileNodeManagerException {
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: line 686 and 687 can be concatenated

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java [hunk 3]
```diff
@@ -671,11 +710,11 @@ public boolean appendFileToFileNode(String fileNodeName, IntervalFileNode append
...
    *
    * @param fileNodeName the seriesPath of storage group
-   * @param appendFile   the appended tsfile information
+   * @param appendFile the appended tsfile information
    */
   public List<String> getOverlapFilesFromFileNode(String fileNodeName, IntervalFileNode appendFile,
-                                                  String uuid)
-          throws FileNodeManagerException {
+      String uuid)
+      throws FileNodeManagerException {
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: concat line 716 717

File: iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java [hunk 4]
```diff
@@ -94,4 +100,59 @@ public TimeValuePairSorter query(String deviceId, String measurement, TSDataType
...
+   * data. Otherwise return null.
+   *
+   * @param chunk the source chunk.
+   * @param timestamp the upper-bound of deletion time.
+   * @return A reduced copy of chunk if chunk contains data with timestamp less than 'timestamp', of
+   * null.
+   */
+  private IWritableMemChunk filterChunk(IWritableMemChunk chunk, long timestamp) {
+    List<TimeValuePair> timeValuePairs = chunk.getSortedTimeValuePairList();
+    if (timeValuePairs.size() > 0 && timeValuePairs.get(0).getTimestamp() <= timestamp) {
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java]: use `isEmpty()` to check

File: iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java [hunk 5]
```diff
@@ -94,4 +100,59 @@ public TimeValuePairSorter query(String deviceId, String measurement, TSDataType
...
+            case INT32:
+              newChunk.putInt(pair.getTimestamp(), pair.getValue().getInt());
+              break;
+            case FLOAT:
+              newChunk.putFloat(pair.getTimestamp(), pair.getValue().getFloat());
+              break;
+            case TEXT:
+              newChunk.putBinary(pair.getTimestamp(), pair.getValue().getBinary());
+              break;
+          }
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java]: Add `default` and throw `UnsupportedOperationException` could remind developers of potential bugs when new `TSDataType` is added

File: iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/IWritableMemChunk.java [hunk 6]
```diff
@@ -18,7 +18,8 @@
  */
 package org.apache.iotdb.db.engine.memtable;
 
-import org.apache.iotdb.tsfile.utils.Binary;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+    import org.apache.iotdb.tsfile.utils.Binary;
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/IWritableMemChunk.java]: irregular indent

File: iotdb/src/main/java/org/apache/iotdb/db/engine/modification/Deletion.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    return timestamp;
+  }
+
+  public void setTimestamp(long timestamp) {
+    this.timestamp = timestamp;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof Modification)) {
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/modification/Deletion.java]: Referring to some official packages like `String`, maybe we'd add the following check:
```
if (this == anObject) {
     return true;
}
```

File: iotdb/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  public LocalTextModificationAccessor(String filePath) {
+    this.filePath = filePath;
+  }
+  @Override
+  public Collection<Modification> read() throws IOException {
+    BufferedReader reader;
+    try {
+      reader = new BufferedReader(new FileReader(filePath));
+    } catch (FileNotFoundException e) {
+      return null;
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java]: Comparing to `return null`, could it return an empty collection?

File: iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java [hunk 9]
```diff
@@ -192,8 +193,8 @@ public void insert(TSRecord tsRecord) throws IOException {
    */
   @Deprecated
   public void update(String deviceId, String measurementId, long startTime, long endTime,
-                     TSDataType type,
-                     byte[] value) {
+      TSDataType type,
+      byte[] value) {
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java]: concat line

File: iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java [hunk 10]
```diff
@@ -203,10 +204,10 @@ public void update(String deviceId, String measurementId, long startTime, long e
    */
   @Deprecated
   public void update(String deviceId, String measurementId, long startTime, long endTime,
-                     TSDataType type,
-                     String value) {
+      TSDataType type,
+      String value) {
```
Comment (kr11) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java]: concat line

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java (hunk 1 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java (hunk 2 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java (hunk 3 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java (hunk 4 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java (hunk 5 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/IWritableMemChunk.java (hunk 6 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/IWritableMemChunk.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/modification/Deletion.java (hunk 7 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/modification/Deletion.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java (hunk 8 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java (hunk 10 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java (hunk 9 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java]: dixed

**Comment by lanking520** [comment]
Please try to break the large commit into small segement for reviewing purpose. Such as 
1. codestyle fix
2. new feature addition
3. bug fixes

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java (hunk 1 above)
Comment (jixuan1989) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: > fixed

Are you sure?  I can still find many many wrong indent ...

Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: Some problems are inherited from the master branch, and checking every file introduces great overhead.

File: iotdb/src/main/java/org/apache/iotdb/db/engine/modification/Modification.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    this.type = type;
+  }
+
+  public enum Type {
+    DELETION
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof Modification))
```
[the remark this answers is not in the document]
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/modification/Modification.java]: fixed

**Pull request merged**